### DISCUSSION TITLE
Issues with Committing to Remote

### DISCUSSION THREAD

**Discussion by gsomaratne** [CS342/.github#17] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

I finished up my view and committed one more time to my feature branch but when I tried to push to remote it won't let me.
<img width="843" alt="Screenshot 2023-01-24 at 12 18 34 PM" src="https://user-images.githubusercontent.com/73210727/214409546-e719eea3-9e4e-483d-9ca0-8a171aa38911.png">

### Reproduction

<img width="1440" alt="Screenshot 2023-01-24 at 12 18 43 PM" src="https://user-images.githubusercontent.com/73210727/214409554-829c6045-25fc-4e3f-bcd2-6a75a773b0dc.png">

### Expected behavior

I expected that without checking ammend commit I could push to remote but I can't.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @gsomaratne,

Thank you for your question!
You should be able to push any changes to the remove (GitHub) using the "Source Control > Push" menu in Xcode. You can find the instructions on that here at the bottom of the documentation: https://developer.apple.com/documentation/xcode/configuring-your-xcode-project-to-use-source-control

The screenshot you have shown doesn't allow you to commit (and then push your files) as you have not selected any files to commit in the list of files above. You should be able to check all the checkboxes next to the files, then check "push to remove: " and then be able to commit + push your changes using the "Commit Files" button.

You can also add, commit, and push your changes using the command line. This tutorial and example provides a good overview of that: https://docs.github.com/en/get-started/using-git/about-git#example-contribute-to-an-existing-repository 👍

Let us know if this solves your problem!
